Extension/src/LanguageServer/settingsPanel.ts

```diff
@@ -0,0 +1,213 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as path from 'path';
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+import * as util from '../common';
+import * as config from './configurations';
+
+// TODO: share ElementId between SettingsPanel and SettingsApp. Investigate why SettingsApp used for HTML cannot import/export
+const ElementId: { [key: string]: string } = {
+    ActiveConfig: "activeConfig",
+    CompilerPath: "compilerPath",
+    IntelliSenseMode: "intelliSenseMode", 
+    IncludePath: "includePath",
+    Defines: "defines",
+    cStandard: "cStandard",
+    cppStandard: "cppStandard"
+};
+
+export interface ViewStateEvent {
+    isActive: boolean;
+}
+
+export class SettingsPanel {
+
+    private configValues: config.Configuration;
+    private configDirty: boolean = false;
+    private settingsPanelViewStateChanged = new vscode.EventEmitter<ViewStateEvent>(); 
+    private panel: vscode.WebviewPanel;
+    private disposable: vscode.Disposable = undefined;
+    private disposablesPanel: vscode.Disposable = undefined;
+
+    constructor() {
+        this.configValues = { name: undefined };
+        this.disposable = vscode.Disposable.from(this.settingsPanelViewStateChanged);
+    }
+
+    public CreateOrShow(activeConfiguration: config.Configuration): void {

```

Comment:
style: All function names should start with a lower-case letter.  Please update the functions in this class.

Reply:
Does this apply to getters and setters?

Reply:
Saw in other classes that the getters and setters start with upper-case